src/Cake.Core/IO/NuGet/INuGetPackageAssembliesLocator.cs

```diff
@@ -0,0 +1,19 @@
+using System.Runtime.Versioning;
+
+namespace Cake.Core.IO.NuGet
+{
+    /// <summary>
+    /// Finds assemblies included in a nuget package.
+    /// </summary>
+    public interface INuGetPackageAssembliesLocator
+    {
+        /// <summary>
+        /// Finds assemblies (DLLs) included in a nuget package.
+        /// </summary>
+        /// <param name="packageDirectory">The package directory.</param>
+        /// <returns>
+        /// the DLLs as IFile[].
+        /// </returns>
+        IFile[] FindAssemblies(DirectoryPath packageDirectory);
```

Comment:
Would prefer if we returned a `IReadOnlyList<IFile>` from public API's that returns a materialized list.